ext/geos_c_impl/factory.c

```diff
@@ -831,10 +832,21 @@ rgeo_convert_to_geos_geometry(VALUE factory, VALUE obj, VALUE type)
     object =
       rb_funcall(rgeo_feature_module, rb_intern("cast"), 3, obj, factory, type);
   }
-  if (NIL_P(object))
+  if (NIL_P(object)) {
+    msg = rb_str_new_cstr("Unable to cast the geometry to the GEOS Factory");
+    rb_protect(
+      rb_exc_raise, rb_exc_new_str(rb_eRGeoInvalidGeometry, msg), state);
```

Comment:
IDK if you had an issue, but you could avoid the temp ruby variable

```suggestion
    rb_protect(
      rb_exc_raise, rb_exc_new_cstr(rb_eRGeoInvalidGeometry, "Unable to cast the geometry to the GEOS Factory"), state);
```

Reply:
Ahh completely missed that there's a `cstr` version of the function. I'll change to that. Tried using a cstr in the function I used and it would compile, but segfaulted once it got to that line.